### PR TITLE
Migrate from Contacts API to People API, enable marking messages as "read" in Inbox

### LINKED ISSUE #4106: Migrate from Contacts API to People API
err 400 when calling https://www.googleapis.com/m8/feeds/contacts/default/thin : _"Contacts API is being deprecated. Migrate to People API to retain programmatic access to Google Contacts. See https://developers.google.com/people/contacts-api-migration."_

Instead we need to use https://developers.google.com/people/v1/other-contacts

This will require oAuth verification of new scope `https://www.googleapis.com/auth/contacts.other.readonly`. Typically, Google will want a video of the new functionality already implemented before allowing new scope.

### PR DESCRIPTION
**Pull request by limonte**
This PR migrates contacts search from deprecated Contacts API to People API

close #4106

----------------------------------

**Tests** _(delete all except exactly one)_:
- Does not need tests (refactor only), but mock endpoints are updated

--------------------------------

### To be filled by reviewers

I have reviewed that this PR... _(tick whichever items you personally focused on during this review)_:
- [ ] addresses the issue it closes (if any)
- [ ] code is readable and understandable
- [ ] is accompanied with tests, or tests are not needed
- [ ] is free of vulnerabilities
- [ ] is documented clearly and usefully, or doesn't need documentation

### CODE REVIEW

**Comment by limonte** [comment]
@tomholub I would need your help here. I'm on the step 3 from your request:

> 3. you will see a big warning about scopes being wrong - continue anyway

I do see the big warning, but I'm unable to continue from it:

CleanShot 2021-11-16 at 23 38 54@2x

I tried to enable [less secure apps access](https://myaccount.google.com/lesssecureapps), but that didn't help. Are you experiencing the same for your account? Do you maybe have any idea how to suppress this warning?

**Comment by tomholub** [comment]
Did you also try on your flowcrypt.com account? In case it only allows for our own org.

**Comment by limonte** [comment]
> Did you also try on your flowcrypt.com account? In case it only allows for our own org.

Yup, same result.

**Comment by tomholub** [comment]
Looking into google developer console -  maybe this time it cannot be done. I'll submit the app for verification, maybe Google has a process for this.

Meanwhile you can continue integration blindly. Can also write a mock api in `google-endpoints.ts` and integrate against that. Then later you'll only have to adjust it once the API can be live-tested.

image

**Comment by tomholub** [comment]
Please also help me make a video demonstrating today's contacts lookup (from app setup to composing new email to looking up contacts and sending the email). Thanks!

**Comment by limonte** [comment]
@tomholub hopefully, this one would do the work, but let me know if I missed anything:

https://user-images.githubusercontent.com/6059356/142082056-1d423d63-9941-4e41-9469-324c9da32f72.mp4

**Comment by tomholub** [comment]
Thank you - this is it

**Comment by tomholub** [comment]
Submitted for verification of the new scopes

File: extension/manifest.json
Comment (tomholub) [inline extension/manifest.json:38]: Dangerous - this will likely cause all existing extensions to get disabled after update.

I forgot why we need this. Is it CORS? Is it to call them from content scripts?

Also needed if we relay requests through background page?

Comment (limonte) [reply extension/manifest.json:38]: @tomholub Sorry for the confusion @tomholub , the PR is still WIP. I'm actually not sure if this change is needed, I was planning to verify that after the verification from Google is done.

Comment (tomholub) [reply extension/manifest.json:38]: Got it

**Comment by limonte** [comment]
@tomholub ready for review 🚀 

Here's the video for the verification:

https://user-images.githubusercontent.com/6059356/142698368-5c8bfd3f-7bf0-469a-84cb-7fc0be9cfb8b.mp4

Hopefully I didn't forget anything, but let me know if I did, thanks!

File: extension/manifest.json
Comment (limonte) [reply extension/manifest.json]: this change wasn't needed after all

File: extension/js/common/api/email-provider/gmail/google.ts
Comment (rrrooommmaaa) [inline extension/js/common/api/email-provider/gmail/google.ts:66]: do we always expect to have a primary name when we have a primary emailAddress?
Otherwise crash this function?

Comment (tomholub) [reply extension/js/common/api/email-provider/gmail/google.ts:66]: I would not force unwrap `!` any data that comes from external API. As Roman says, it may break things.

Comment (limonte) [reply extension/js/common/api/email-provider/gmail/google.ts]: Right, done 👍

**Comment by tomholub** [comment]
Very good - I'll submit verification video and wait for news. Once approved, I'll merge it.

**Comment by tomholub** [comment]
Submitted necessary details go Google - now waiting for their response.

**Review by tomholub — APPROVED** [review]

**Pull request merged**
